src/lib.rs

```diff
@@ -394,6 +394,7 @@ authors has done substantial research on the integration of CSP and programming
 languages.
 */
 
+#![deny(warnings)]
```

Comment:
I admire your gusto, but we shouldn't fail the build because there are warnings. If new warnings come out again, then `chan` will stop compiling and that's worse than spewing warnings but still compiling.

Reply:
As a compromise, you could do `#![cfg_attr(test, deny(warnings))]`, which will fail the build only in tests.